### PR TITLE
Adds Suit Storage Tgui

### PR DESCRIPTION
**Pull request by AnturK**

### CODE REVIEW

**Comment by ChangelingRain** [comment]
`controls buttons and meters` should be `controls, buttons, and meters`

**Comment by AnturK** [comment]
Hum, weird i updated that in the template

**Comment by optimumtact** [comment]
\"e\" ??

**Comment by neersighted** [comment]
Looks pretty good. I'd loose the 'bad' in the ui-notice; and drop the flavor text -- its too long for a ui-notice.

**Comment by JJRcop** [comment]
Hey, we could just get a stock dial image and use transforms on it, then have the LEDs, then the background can be a tileable pipe mess

**Comment by vista-** [comment]
The weird symbol is probably meant to be λ

File: code/game/machinery/suit_storage_unit.dm [hunk 1]
```diff
@@ -185,104 +186,62 @@
...
+	var/list/data = list()
+	data["isBroken"] = isbroken
+	data["isLocked"] = islocked
+	data["isOpen"] = isopen
+	data["isBaking"] = isUV
+	data["uv"] = issuperUV
+	data["safety"] = safetieson
+	data["maintenance"] = maintenance_mode
+	if(HELMET)
+		data["helmet"] = HELMET.name
```
Comment (neersighted) [inline code/game/machinery/suit_storage_unit.dm]: Loose the all caps vars

File: code/game/machinery/suit_storage_unit.dm [hunk 2]
```diff
@@ -185,104 +186,62 @@
...
+		if ("uv_start")
+			start_UV(usr)
+		if ("mob")
+			eject_occupant(usr)
+	update_icon()
+	return 1
 
 /obj/machinery/suit_storage_unit/proc/toggleUV(mob/user)
-	if(!src.panel_open)
+	if(!src.maintenance_mode)
```
Comment (neersighted) [inline code/game/machinery/suit_storage_unit.dm]: Loose src

File: code/game/machinery/suit_storage_unit.dm [hunk 3]
```diff
@@ -325,60 +284,59 @@
...
-	src.HELMET = null
-	src.MASK = null
-	src.STORAGE = null
-	if(src.OCCUPANT)
-		src.eject_occupant(OCCUPANT)
+	SUIT = null
+	HELMET = null
+	MASK = null
+	STORAGE = null
+	if(OCCUPANT)
```
Comment (neersighted) [inline code/game/machinery/suit_storage_unit.dm]: All caps is bad

File: code/game/machinery/suit_storage_unit.dm [hunk 4]
```diff
@@ -386,54 +344,53 @@
...
 /obj/machinery/suit_storage_unit/proc/cycletimeleft()
-	if(src.cycletime_left >= 1)
-		src.cycletime_left--
-	return src.cycletime_left
+	if(cycletime_left >= 1)
+		cycletime_left--
+	return cycletime_left
 
 
 /obj/machinery/suit_storage_unit/proc/eject_occupant(mob/user)
```
Comment (neersighted) [inline code/game/machinery/suit_storage_unit.dm]: Lots of all caps

File: code/game/machinery/suit_storage_unit.dm [hunk 5]
```diff
@@ -476,13 +432,13 @@
...
-	if (!src.isopen)
+	if (!isopen)
 		user << "<span class='warning'>The unit's doors are shut!</span>"
 		return
-	if (!src.ispowered || src.isbroken)
+	if (!ispowered || isbroken)
 		user << "<span class='warning'>The unit is not operational!</span>"
 		return
-	if ( (src.OCCUPANT) || (src.HELMET) || (src.SUIT) || (src.STORAGE))
+	if ( (OCCUPANT) || (HELMET) || (SUIT) || (STORAGE))
```
Comment (neersighted) [inline code/game/machinery/suit_storage_unit.dm]: Loose the extra parens

File: tgui/src/interfaces/suit_storage.ract [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  </ui-notice>
+{{/if}}
+
+{{#if data.isBaking}}
+  <ui-notice>
+    <span>Unit is cauterising contents with selected UV ray intensity. Please wait.</span>
+  </ui-notice>
+{{/if}}
+
+{{#if data.occupied}}
```
Comment (neersighted) [inline tgui/src/interfaces/suit_storage.ract]: I would also lose the all caps warning

File: tgui/src/interfaces/suit_storage.ract [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      </ui-section>
+  {{/partial}}
+  {{#if data.isLocked}}
+    <ui-notice>
+      <span>LOCKED</span>
+    </ui-notice>
+  {{else}}
+      {{#if data.maintenance}}
+          <ui-notice>
+            <span>The panel is ridden with controls{{','}} buttons{{','}} and meters{{','}} labeled in strange signs and symbols that you cannot understand; probably the manufacturing world's language. Among other things, a few controls catch your eye...</span>
```
Comment (neersighted) [inline tgui/src/interfaces/suit_storage.ract]: This is fluff text and does not belong in a UI. Ideally, remove this in favor of a wire datum.

File: tgui/src/interfaces/suit_storage.ract [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      </ui-section>
+  {{/partial}}
+  {{#if data.isLocked}}
+    <ui-notice>
+      <span>LOCKED</span>
+    </ui-notice>
+  {{else}}
+      {{#if data.maintenance}}
+          <ui-section>
+            <div>A small dial with a &lambda; symbol engraved on it.</div>
```
Comment (neersighted) [inline tgui/src/interfaces/suit_storage.ract]: Still fluff text. Oh well, I'll port it to wires later. This is good for now.

**Comment by neersighted** [comment]
I'm gonna go ahead and :+1: this. I don't like maint mode, I'll rewrite it as a wire datum later.

**Pull request merged**